### PR TITLE
Observable preferences I (Internal [formerly Version], Groups, Xmp, AutoComplete)

### PR DESCRIPTION
**Pull request by calixtus**
Follow up to #8260 

Note: enabling autocomplete without restarting probably requires a major refactor of the entry editor. As I believe that this would escalate this PR too much, I think we should leave this for later, maybe when the entry editor is redesigned.

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/gui/groups/GroupTreeView.java [hunk 1]
```diff
@@ -199,9 +202,17 @@ private void initialize() {
                                 group.allSelectedEntriesMatchedProperty());
                     }
                     Text text = new Text();
-                    if (preferencesService.getDisplayGroupCount()) {
-                        text.textProperty().bind(group.getHits().asString());
-                    }
+                    EasyBind.subscribe(preferencesService.getGroupsPreferences().displayGroupCountProperty(),
+                            (newValue) -> {
+                                if (text.textProperty().isBound()) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/groups/GroupTreeView.java:207]: Why is this necessary? Please add a comment

Comment (calixtus) [reply src/main/java/org/jabref/gui/groups/GroupTreeView.java:207]: I cannot set the text to an empty value if the value is bound, instead it keeps the last calculated number of entries in `group.getHits()` or just `0`, if the preference option to calculate the number of entries (which is very expensive in large databases) is deactivated. If the calculation of entries is disabled and you add or remove an entry from a group it would display the wrong number of entries.

**Review by Siedlerchr — APPROVED** [review]
